Use env. var. CFLAGS & LDFLAGS to compile with hardening options
--- a/src/qasm-simulator-cpp/src/Makefile
+++ b/src/qasm-simulator-cpp/src/Makefile
@@ -21,9 +21,9 @@
 # Check if compiler is Apple LLVM and doesn't support OpenMP
 APPLELLVM = $(shell ${CC} --version | grep ^Apple)
 ifeq ($(APPLELLVM),)
-	CPPFLAGS = $(USER_FLAGS) -std=c++11 -fopenmp $(OPT) $(INCLUDE) $(WARNINGS)
+	CPPFLAGS = $(USER_FLAGS) -std=c++11 -fopenmp $(OPT) $(INCLUDE) $(WARNINGS) $(CFLAGS) $(LDFLAGS)
 else
-	CPPFLAGS = $(USER_FLAGS) -std=c++11 $(OPT) $(INCLUDE) $(WARNINGS)
+	CPPFLAGS = $(USER_FLAGS) -std=c++11 $(OPT) $(INCLUDE) $(WARNINGS) $(CFLAGS) $(LDFLAGS)
 endif
 
 # BLAS distribution to use (change to use OpenBLAS or MKL for example)
